s3/url_test.go

```diff
@@ -0,0 +1,38 @@
+package s3_test
```

Comment:
Torn on this one, I've not generally seen the package redefined within itself. And if this stayed as s3 you wouldn't need to import the s3 package. But I suppose that means the TestSpec method is exported as part of the s3 package.

This currently breaks the convention set by other tests in this repo, but may be worth a discussion around rather than assuming they're correct by default.